### PR TITLE
Add channel property to Version

### LINKED ISSUE #3466: Add `channel` field to `version` and set it on version submission
Part of #3465

This is only adding the `channel` column to the `version` entity and setting it with the same value as `is_listed` from `addon` when a new version is uploaded. So, Unlisted would be 0 and Listed would be 1. Note that we might need a third channel in the future, so the column should be numerical.

### PR DESCRIPTION
**Pull request by eviljeff**
Minimally fixes #3466

### CODE REVIEW

File: src/olympia/versions/models.py [hunk 1]
```diff
@@ -92,6 +92,9 @@ class Version(OnChangeMixin, ModelBase):
     source = models.FileField(
         upload_to=source_upload_path, null=True, blank=True)
 
+    channel = models.IntegerField(choices=amo.RELEASE_CHANNELS,
```
Comment (diox) [inline src/olympia/versions/models.py]: That's not a valid `choices`. django system checks complain about it:
`versions.Version.channel: (fields.E005) 'choices' must be an iterable containing (actual value, human readable name) tuples.`

**Comment by diox** [comment]
Can you also add a couple tests making sure that `channel` is set properly when we upload something via devhub or via the signing API ? I know the unit tests you added should be enough, but...

File: src/olympia/versions/models.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/versions/models.py]: weird that the tests don't trigger that error

Comment (eviljeff) [inline src/olympia/versions/models.py]: fixed choices. 
Just going add those tests...

**Comment by eviljeff** [comment]
Added to existing tests.  Is that sufficient?

**Comment by diox** [comment]
Yeah that's perfect.

**Pull request merged**